templates/EmptyModel.cpp

```diff
@@ -30,74 +28,35 @@ EmptyModel::EmptyModel(const ConfigMap& configMap) : EmptyModel::EmptyModel() {
 }
 
 // EmptyModel State Equation
-std::vector<double> EmptyModel::stateEqn(const double,
-                                         std::vector<double>& x,
-                                         const std::vector<double>& u,
-                                         const std::vector<double>& n,
-                                         const double dt) const {
-    // Extract states
-    // double a = x[0];
-    // double b = x[1];
-    // ...
-
-    // Extract inputs
-    // double c = u[0];
-    // ...
-
-    // State equations
-    // double adot = a + b;
-    // double bdot = c;
-    // ...
-
-    // Update state
-    // x[0] = a + adot*dt;
-    // x[1] = b + bdot*dt;
-    // ...
-
-    // Add process noise
-    // x[0] += dt*n[0];
-    // x[1] += dt*n[1];
-    // ...
+Model::state_type EmptyModel::stateEqn(double t,
+                                         const Model::state_type& x,
+                                         const Model::input_type& u,
+                                       const Model::noise_type& n,
+                                         double dt) const {
+    Model::state_type new_state;
+    
+    // Fill new_state
+    
+    return new_state;
 }
 
 // EmptyModel Output Equation
-std::vector<double> EmptyModel::outputEqn(const double,
-                                          const std::vector<double>& x,
-                                          const std::vector<double>&,
-                                          const std::vector<double>& n,
-                                          const std::vector<double>& z) const {
-    // Extract states
-    // double a = x[0];
-    // double b = x[1];
-    // ...
-
-    // Extract inputs
-    // double c = u[0];
-    // ...
-
-    // Output equations
-    // double d = a + b + c;
-    // ...
-
-    // Set outputs
-    // z[0] = d;
-    // ...
-
-    // Add noise
-    // z[0] += n[0];
-    // ...
+Model::output_type EmptyModel::outputEqn(double t,
+                                         const Model::state_type& x,
+                                         const Model::input_type& u,
+                                         const Model::noise_type& n) const {
+    Model::output_type output;
```

Comment:
These types aren't automatically sized, so this line should be `auto output = getOutputVector();` to get an output vector of the correct size. Maybe we should delete the default constructor of `DynamicArray` to make sure this mistake isn't possible.